test/utilities.jl

```diff
@@ -45,7 +45,7 @@ end
 		r = only(randn(rng, 1));
 	end
 
-	@test all(sort!(single_thread) .== sort!(multi_thread))
+	@test all(sort!(single_thread) .≈ sort!(multi_thread))
```

Comment:
Seems good to me. Worth specifying a tolerance explicitly do you think, or is the default sufficient? Also, you can do this if you feel so inclined:
```suggestion
	@test sort!(single_thread) ≈ sort!(multi_thread)
```
since there is an `isapprox` for arrays, though it uses `norm` rather than elementwise approximate equality, IIRC.

Reply:
I thought about that and think I want the elementwise comparison because our guarantee is on the elementwise equality.

I suspect the default tolerance is good enough -- it should reflect what floating-point equality can reasonably be expected to be, right? I definitely don't want a looser tolerance, but I have no idea how tight we can go -- 1 ups? 2 ups? 1e-16? I have no idea.